realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -136,6 +146,403 @@ public boolean load() {
         return true;
     }
 
+
+    /**
+     * Updates the field given by {@code fieldName} in all objects inside the query result.
+     * <p>
+     * This method will automatically try to convert numbers and booleans that are given as
+     * {@code String} to their appropriate type. For example {@code "10"} will be converted to
+     * {@code 10} if the field type is {@link RealmFieldType#INTEGER}.
+     * <p>
+     * Using the typed setters like {@link #setInt(String, int)} will be faster than using
+     * this method.
+     *
+     * @param fieldName field to update
+     * @param value value to update with.
+     * @throws IllegalArgumentException if the field could not be found, could not be updated or
+     * the argument didn't match the field type or could not be converted to match the underlying
+     * field type.
+     */
+    public void setValue(String fieldName, @Nullable Object value) {
+        checkNonEmptyFieldName(fieldName);
+        fieldName = mapFieldNameToInternalName(fieldName);
+        realm.checkIfValid();
```

Comment:
You should call `checkIfValidAndInTransaction` instead (same for all `set*` below)